src/HealthChecks/HealthChecks/src/HealthCheckService.cs

```diff
@@ -38,6 +39,7 @@ public abstract class HealthCheckService
         /// A <see cref="Task{T}"/> which will complete when all the health checks have been run,
         /// yielding a <see cref="HealthReport"/> containing the results.
         /// </returns>
+        [SuppressMessage("ApiDesign", "RS0026:Do not add multiple public overloads with optional parameters", Justification = "Required to maintain compatibility")]
         public Task<HealthReport> CheckHealthAsync(CancellationToken cancellationToken = default)
```

Comment:
nit: I wish the analyzer would suppress itself when the only optional parameter is a `CancellationToken`. The "overload for `int`, `float`, &hellip; cases e.g. `WriteAsync(...)` methods and similar are causing a lot of suppression noise.

@sharwell @mavasani @jcouv❔